tests/io/test_scans.py

```diff
@@ -2059,3 +2059,42 @@ def test_scan_create_scan_document_credentials_pass(api):
     for credential in credentials:
         getattr(api.scans, '_create_scan_document')(
             {'credentials': credential})
+
+
+@pytest.mark.vcr()
+def test_block_while_running(api):
+    '''
+    test _block_while_running with expected type.
+    '''
+    api.scans._block_while_running(scan_id=583, sleeper=5)
+
+
+@pytest.mark.vcr()
+def test_scan_results_scan_notfounderror(api):
+    '''
+    test to raise exception when scan not found.
+    '''
+    with pytest.raises(NotFoundError) as not_found_error:
+        api.scans.results(scan_id=583, history_id=12, history_uuid="123e4567-e89b-12d3-a456-426614174000")
+    assert "Scan not found" in not_found_error.value.msg, \
+        "Invalid type validation error for scan_id parameter is not raised by test-case."
+
+
+@pytest.mark.vcr()
+def test_scan_history_sort_direction(api):
+    '''
+    test scan history for sort direction.
+    '''
+    api.scans.history(scan_id=11, limit=10, offset="1", pages=3, sort=(('name', 'asc'), ('description', 'desc')))
```

Comment:
The `history()` function returns a `ScanHistoryIterator`. Is there a reason for skipping the assertion?

Reply:
In positive testcase we followed structure of existing testcase (`test_scan_create_scan_document_credentials_pass`)